src/gui/symbology-ng/qgssymbolv2selectordialog.cpp

```diff
@@ -45,6 +45,65 @@
 
 static const int SymbolLayerItemType = QStandardItem::UserType + 1;
 
+DataDefinedRestorer::DataDefinedRestorer( QgsSymbolV2* symbol, const QgsSymbolLayerV2* symbolLayer )
+  : mMarker( NULL )
+  , mMarkerSymbolLayer( NULL )
+  , mLine( NULL )
+  , mLineSymbolLayer( NULL )
+{
+  if ( symbolLayer->type() == QgsSymbolV2::Marker )
+  {
+    Q_ASSERT( symbol->type() == QgsSymbolV2::Marker );
+    mMarker = static_cast<QgsMarkerSymbolV2*>( symbol );
+    mMarkerSymbolLayer = static_cast<const QgsMarkerSymbolLayerV2*>( symbolLayer );
+    mDDSize = mMarker->dataDefinedSize();
+    mDDAngle = mMarker->dataDefinedAngle();
+  }
+  else if( symbolLayer->type() == QgsSymbolV2::Line )
+  {
+    Q_ASSERT( symbol->type() == QgsSymbolV2::Line );
```

Comment:
@nyalldawson Is the crash you're referring to the failure of this assertion ? If it's the case, replacing the assertion by and && in the if will do the trick, won't it ?

I was not conscious of the line layer on Fill symbols. Thanks